lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -15,44 +15,59 @@ class XpayGateway < Gateway
 
       ENDPOINTS_MAPPING = {
         purchase: 'orders/2steps/payment',
+        authorize: 'orders/2steps/payment',
         preauth: 'orders/2steps/init',
-        capture: 'operations/{%s}/captures',
+        capture: 'operations/%s/captures',
         verify: 'orders/card_verification',
-        void: 'operations/{%s}/cancels',
-        refund: 'operations/{%s}/refunds'
+        void: 'operations/%s/cancels',
+        refund: 'operations/%s/refunds'
       }
 
+      module PaymentActions
+        PREAUTH = :preauth
+        AUTHORIZE = :authorize
+        PURCHASE = :purchase
+        CAPTURE = :capture
+        VERIFY = :verify
+        VOID = :void
+        REFUND = :refund
+      end
```

Comment:
This seems like overkill for this scenario. I think for your `case/when` statement, you can just check the `action` type and make your decision based on that `action` key to set the `captureType`

Reply:
Understood! Corrected!